### PR TITLE
delay_span_bug instead of silent ignore

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
This is a follow-up to #74557.

r? @pnkfelix

### CODE REVIEW

File: src/librustc_resolve/late.rs [hunk 1]
```diff
@@ -1508,16 +1508,22 @@ impl<'a, 'b, 'ast> LateResolutionVisitor<'a, 'b, 'ast> {
             match pat.kind {
                 // In tuple struct patterns ignore the invalid `ident @ ...`.
                 // It will be handled as an error by the AST lowering.
```
Comment (estebank) [inline src/librustc_resolve/late.rs:1510]: This comment should have gone with the `delay_span_bug` :)

**Comment by Mark-Simulacrum** [comment]
@bors r=pnkfelix

Fixed @estebank's nit, but not going to backport that to stable (not worth the hassle).

**Comment by cuviper** [comment]
Should this also be beta-nominated to follow #74557? Especially since this was backported to stable 1.45.1, but maybe not depending the investigation of #74954.

**Pull request merged**